app/src/main/java/com/mashup/mobalmobal/ui/sign/signin/SignInFragment.kt

```diff
@@ -1,25 +1,147 @@
+import com.mashup.mobalmobal.databinding.FragmentSignInBinding
+import com.mashup.mobalmobal.util.showToastMessage
+import dagger.hilt.android.AndroidEntryPoint
 
-/**
- * 로그인
- */
+
+@AndroidEntryPoint
 class SignInFragment : BaseFragment() {
+
+    companion object {
+        private const val TAG = "SignInFragment"
+    }
+
+    private var _binding: FragmentSignInBinding? = null
+    private val binding get() = _binding!!
+
+    private val viewModel by viewModels<SignInViewModel>()
+
+    private lateinit var googleSignInClient: GoogleSignInClient
+    private lateinit var googleLoginLauncher: ActivityResultLauncher<Intent>
+
+    private lateinit var callbackManager: CallbackManager
+
+
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
-        return inflater.inflate(R.layout.fragment_sign_in, container, false)
+    ): View {
+        _binding = FragmentSignInBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        setGoogleLogin()
+        setFacebookLogin()
+
+        binding.signInGoogle.setOnClickListener {
+            onGoogleClicked()
+        }
+
+        viewModel.showToast.observe(viewLifecycleOwner, { message ->
+            showToastMessage(message)
+        })
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        callbackManager.onActivityResult(requestCode, resultCode, data)
+    }
+
+    private fun setGoogleLogin() {
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestIdToken(getString(R.string.default_web_client_id))
+            .requestEmail()
+            .build()
+
+        googleSignInClient = GoogleSignIn.getClient(requireActivity(), gso)
+
+        googleLoginLauncher = registerForActivityResult(GoogleLoginResultContract()) { idToken ->
+            idToken?.let {
+                viewModel.handleGoogleAccessToken(it)
+            }
+        }
+    }
+
+    private fun setFacebookLogin() {
+        callbackManager = CallbackManager.Factory.create()
+
+        binding.signInFacebook.setPermissions("email", "public_profile")
+        binding.signInFacebook.fragment = this
+        binding.signInFacebook.registerCallback(
+            callbackManager,
+            object : FacebookCallback<LoginResult> {
+                override fun onSuccess(result: LoginResult?) {
+                    result?.let {
+                        viewModel.handleFacebookAccessToken(it.accessToken)
+                    }
+                }
+
+                override fun onCancel() {
+                }
```

Comment:
```suggestion
                override fun onCancel() { }
```
이거 꼭 추가해야 하는 메소드인가요?

Reply:
네 지우면 빨간줄 뜨네요 ....